core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1260,16 +1261,19 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
             }
 
             QualifiedObjectName name = createQualifiedObjectName(session, table, table.getName());
-            analysis.addEmptyColumnReferencesForTable(accessControl, session.getIdentity(), name);
```

Comment:
I think we can leave `addEmptyColumnReferencesForTable` as is.

Reply:
Could you elaborate why we'd want to keep it that way? 

my reasoning was that: `tableColumnReferences` are used for ACL checks.  in case of views/materialized views, that should be the source, but for tables, that should be redirected table. so separating them makes this more readable.

If we consider that column metadata responsibility is always on target, the currently implemented behavior would make sense right?